subt/tools/analyze_detectors.py

```diff
@@ -0,0 +1,243 @@
+        true_score_t = []
+        true_score_cv = []
+        false_score_t = []
+        false_score_cv = []
+        for artf, score_t, score_cv, detection_type in data:
+            if detection_type == "true" and artf_name == artf:
+                true_score_t.append(score_t)
+                true_score_cv.append(score_cv)
+            if detection_type == "false" and artf_name == artf:
+                false_score_t.append(score_t)
+                false_score_cv.append(score_cv)
+
+        fig = plt.figure(figsize=(7, 6))
+        fig.subplots_adjust(left=0.1, right=0.8, bottom=0.1)
+
+        ax1 = fig.add_subplot(111)
+        ax1.plot(true_score_t, true_score_cv, "go", label="True")
+        ax1.plot(false_score_t, false_score_cv, "ro", label="False")
+        ax1.grid()
+        ax1.set_title(artf_name)
+        ax1.set_xlabel("mdnet (-)")
+        ax1.set_ylabel("cv detector (-)")
+        ax1.legend(bbox_to_anchor=(1.05, 0.7))
+
+        graph_name = artf_name + "_graph"
+        plt.savefig(graph_name, dpi=500)
+        plt.show()
+
+
+def filter_detections(path):
+    # collect data with detection
+    data_to_filter = []
+    dir_list = sorted(os.listdir(path))
+    for dir in dir_list:
+        if not dir.endswith(".log.d"):
+            continue
+        for line in open(os.path.join(path, dir, "detection.log")):
+            __, artf_name, score_t, score_cv, points, bbox, im_name = line.split(";")
+            score_t = float(score_t)
+            score_cv = float(score_cv)
+            points = eval(points)
+            bbox = eval(bbox)
+            im_path = os.path.join(path, dir, im_name)
+            data_to_filter.append([artf_name, score_t, score_cv, points, bbox, im_path, "true"])
+
+    data = manual_sorting(data_to_filter)
+
+    # save data for future analysis
+    with open(os.path.join(path, "detection_overview.csv"), "w") as overview_log:
+        for artf_name, score_t, score_cv, detection_type in data:
+            overview_log.write("%s,%f,%f,%s\r\n" %(artf_name, score_t, score_cv, detection_type))
+
+    plot_data(data)
+
+
+def eval_logs(path):
+    if os.path.isdir(path):
+        logs_list = sorted(os.listdir(path))
+        for ii, log in enumerate(logs_list):
+            if not log.endswith(".log"):
+                continue
+            if "rosout" in log or "server_console" in log:
+                continue
+            log_eval(os.path.join(path, log))
+    else:
+        log_eval(path)
+
+
+if __name__ == "__main__":
+    import argparse
+    parser = argparse.ArgumentParser(description=__doc__)
+    parser.add_argument('logfiles', help='Path to directory wit logfiles')
+    parser.add_argument('--filter', help='Filter false detection', action='store_true')
+#    parser.add_argument('--plot', help='Plot score', action='store_true')
+    args = parser.parse_args()
+
+    path = args.logfiles
+    if args.filter:
+        filter_detections(path)
+#    elif args.plot:
```

Comment:
Should this be commented out?

Reply:
Yes, there is no follow-up code for this parameter yet.